sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java

```diff
@@ -1608,6 +1609,98 @@ public void canCRUDTrustedLaunchVM() {
+            .define(vmName)
+            .withRegion(region)
+            .withNewResourceGroup(rgName)
+            .withExistingPrimaryNetwork(network)
+            .withSubnet("subnet1")
+            .withPrimaryPrivateIPAddressDynamic()
+            .withoutPrimaryPublicIPAddress()
+            .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_20_04_LTS_GEN2)
+            .withRootUsername("Foo12")
+            .withSsh(sshPublicKey())
+            .withNewDataDisk(10, 1, new VirtualMachineDiskOptions().withDeleteOptions(DeleteOptions.DELETE))
+            .withSize(VirtualMachineSizeTypes.STANDARD_DS3_V2)
+            .withNewSecondaryNetworkInterface(this
+                .networkManager
+                .networkInterfaces()
+                .define(nicName)
+                .withRegion(region)
+                .withExistingResourceGroup(rgName)
+                .withExistingPrimaryNetwork(network)
+                .withSubnet("subnet1")
+                .withPrimaryPrivateIPAddressDynamic(), DeleteOptions.DELETE)
+            .withPrimaryNetworkInterfaceDeleteOptions(DeleteOptions.DELETE)
+            .withOSDiskDeleteOptions(DeleteOptions.DELETE)
+            .create();
+
+        Assertions.assertEquals(DeleteOptions.DELETE, vm.osDiskDeleteOptions());
+        Assertions.assertEquals(DeleteOptions.DELETE, vm.primaryNetworkInterfaceDeleteOptions());
+        Assertions.assertTrue(vm.dataDisks().values().stream().allMatch(disk -> DeleteOptions.DELETE.equals(disk.deleteOptions())));
+
+        // update delete options all to DETACH, except for secondary nic
+        vm.update()
+            .withOsDiskDeleteOptions(DeleteOptions.DETACH)
+            .withPrimaryNetworkInterfaceDeleteOptions(DeleteOptions.DETACH)
+            .withDataDiskDeleteOptions(1, DeleteOptions.DETACH)
+            .withNewDataDisk(1)
+            .apply();
+
+        Assertions.assertEquals(DeleteOptions.DETACH, vm.osDiskDeleteOptions());
+        Assertions.assertEquals(DeleteOptions.DETACH, vm.primaryNetworkInterfaceDeleteOptions());
+        // secondary nic delete options remains unchanged
+        Assertions.assertTrue(vm.networkInterfaceIds().stream()
+            .filter(nicId -> !nicId.equals(vm.primaryNetworkInterfaceId())).allMatch(nicId -> DeleteOptions.DELETE.equals(vm.networkInterfaceDeleteOptions(nicId))));
+        Assertions.assertTrue(vm.dataDisks().values().stream().allMatch(disk -> DeleteOptions.DETACH.equals(disk.deleteOptions())));
+
+        NetworkInterface secondaryNic2 =
+            this
+                .networkManager
+                .networkInterfaces()
+                .define(nicName2)
+                .withRegion(region)
+                .withExistingResourceGroup(rgName)
+                .withExistingPrimaryNetwork(network)
+                .withSubnet("subnet2")
+                .withPrimaryPrivateIPAddressDynamic()
+                .create();
+
+        vm.powerOff();
+        vm.deallocate();
```

Comment:
Is deallocate required?

Reply:
Yeah, powerOff() isn't enough for adding a new nic.